great_expectations/profile/user_configurable_profiler.py

```diff
@@ -211,9 +213,56 @@ def build_suite(self) -> ExpectationSuite:
             self.profile_dataset._expectation_suite = ExpectationSuite(suite_name)
 
         if self.semantic_types_dict:
-            return self._build_expectation_suite_from_semantic_types_dict()
+            expectation_suite = self._build_expectation_suite_from_semantic_types_dict()
+        else:
+            expectation_suite = self._profile_and_build_expectation_suite()
+
+        self._send_usage_stats_message()
+
+        return expectation_suite
 
-        return self._profile_and_build_expectation_suite()
+    def _send_usage_stats_message(self):
+        profile_dataset_type: str = str(type(self.profile_dataset))
+        if "Dataset" in profile_dataset_type:
+            profile_dataset_type = "Dataset"
+        elif "Batch" in profile_dataset_type:
+            profile_dataset_type = "Batch"
+        elif "Validator" in profile_dataset_type:
+            profile_dataset_type = "Validator"
+        else:
+            raise ValueError(
```

Comment:
Maybe you could add this error message to the assert statement in the `__init__()` method?

Reply:
@anthonyburdi Can be done, I just wanted to only minimally alter @talagluck 's implementation.